### PR TITLE
Clean up CMake files

### PR DESCRIPTION
**Pull request by e-kwsm**
#### What does this implement/fix? Explain your changes.

* bump CMake minimum required version to 3.7 to utilize [`VERSION_GREATER_EQUAL`](https://cmake.org/cmake/help/v3.7/release/3.7.html#index-0-command:if)
* improve catch-dependency
* fix conditionals
* use early return
* properly use CMake commands

### CODE REVIEW

**Comment by e-kwsm** [comment]
`VERSION_GREATER_EQUAL` is used at https://github.com/rdkit/rdkit/blob/9a04aea918b6b0689b6a68748385806625cb661a/Code/RDBoost/CMakeLists.txt#L23

**Comment by e-kwsm** [comment]
Is the following modification compatible with the current behaviour?

```diff
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,18 +179,18 @@ endif(RDK_BUILD_SWIG_WRAPPERS)
 
 
 if(RDK_BUILD_COORDGEN_SUPPORT)
-    add_definitions(-DRDK_BUILD_COORDGEN_SUPPORT)
-    include_directories(${RDKit_ExternalDir})
+  target_compile_definitions(rdkit_base INTERFACE -DRDK_BUILD_COORDGEN_SUPPORT)
+  target_include_directories(rdkit_base INTERFACE ${RDKit_ExternalDir})
 endif()
 
 if(RDK_BUILD_MAEPARSER_SUPPORT)
-    add_definitions(-DRDK_BUILD_MAEPARSER_SUPPORT)
-    include_directories(${RDKit_ExternalDir})
+  target_compile_definitions(rdkit_base INTERFACE -DRDK_BUILD_MAEPARSER_SUPPORT)
+  target_include_directories(rdkit_base INTERFACE ${RDKit_ExternalDir})
 endif()
 
 if(RDK_USE_URF)
-  ADD_DEFINITIONS("-DRDK_USE_URF")
-  include_directories(${RDKit_ExternalDir}/RingFamilies/RingDecomposerLib/src/RingDecomposerLib)
+  target_compile_definitions(rdkit_base INTERFACE -DRDK_USE_URF)
+  target_include_directories(rdkit_base INTERFACE ${RDKit_ExternalDir}/RingFamilies/RingDecomposerLib/src/RingDecomposerLib)
 endif()
 
 
```

File: Code/GraphMol/MolDraw2D/QTDemo/CMakeLists.txt [hunk 1]
```diff
@@ -8,8 +8,8 @@ find_package(Boost ${RDK_BOOST_VERSION} COMPONENTS program_options iostreams fil
 set(CMAKE_AUTOMOC ON)
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-  ADD_DEFINITIONS("-fPIC")
-  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+  add_compile_options(-fPIC)
+  set(CMAKE_CXX_STANDARD 11)
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/QTDemo/CMakeLists.txt:12]: don't need this line at all any more

File: Docs/Book/C++Examples/CMakeLists.txt [hunk 2]
```diff
@@ -7,7 +7,9 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${RDBASE}/Code/cmake/Modules")
 
 # note that if you haven't installed/built the toolkit with CoordGen, you'll
 # have problems with this.
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall  -std=c++11 -DRDK_BUILD_COORDGEN_SUPPORT=ON" )
+add_definitions("-DRDK_BUILD_COORDGEN_SUPPORT=ON")
+set(CMAKE_CXX_STANDARD 11)
```
Comment (greglandrum) [inline Docs/Book/C++Examples/CMakeLists.txt:11]: don't need this

**Review by greglandrum — APPROVED** [review]
Thanks for the improvement and sorry I forgot to merge it earlier.
I will (if I can) go ahead and directly make the two suggestions I made and merge now.

**Pull request merged**